src/lib.rs

```diff
@@ -469,6 +567,51 @@ impl PurpleICQ {
                         },
                     );
                 }
+
+                // Get the conversation and set the oldest *displayed* messageId.
+                // This is the oldest message that the user can see in the chat window.
+                //
+                // If there is no conversation yet, that is okay. It means that we haven't
+                // seen new messages yet.
+                if let Some(mut conversation) =
```

Comment:
Can't we just move this block up and use the conversation returned by `conversation_joined` (And make `conversation_joined` return its `conversation`)

Reply:
Done!